contracts/truefi2/strategies/CurveYearnStrategy.sol

```diff
@@ -125,7 +125,7 @@ contract CurveYearnStrategy is UpgradeableClaimable, ITrueStrategy {
         amounts[tokenIndex] = totalAmount;
 
         token.safeApprove(address(curvePool), totalAmount);
-        uint256 conservativeMinAmount = calcTokenAmount(totalAmount).mul(999).div(1000);
+        uint256 conservativeMinAmount = calcTokenAmount(totalAmount, true).mul(999).div(1000);
```

Comment:
Our team tries to adhere to test-driven development. When we fix an issue, we typically write unit or integration tests for the fix. This has three benefits:
1. A failing test confirms the original issue exists.
2. Making the test pass confirms that our fix actually resolves the issue.
3. Leaving the test in our CI framework helps us catch regressions when we refactor in the future.

Could you write a unit test to check we're calling the correct `curve().calc_token_amount(amount, true)` under the deposit section? https://github.com/trusttoken/smart-contracts/blob/65b8699c4b44e5e18b4ff733ca4e0ccc52070ff6/test/truefi2/strategies/CurveYearnStrategy.test.ts#L102

I think all that your test case needs to do is use Waffle's `deployMockContract(owner, ICurveJson.abi)` to deploy a mock, pass this in to a new TestCurveStrategy, and check that the correct arguments get called in the mock `ICurve`.